### PR TITLE
feat: lots of inven features and bugfixes

### PR DESCRIPTION
**Pull request by rahulk29**
- feat: optional ingredients
- fix: duplicate recipes
- feat: add command to view recipe contents
- fix: decimals in recipes
- feat: display raw shopping list

### CODE REVIEW

File: client/helper.py [hunk 1]
```diff
@@ -3,8 +3,8 @@
 import toml
 from client import auth
 
-url = "https://inven.fly.dev/"
-#url = "http://localhost:4000/" #(for local testing)
+# url = "https://inven.fly.dev/"
+url = "http://localhost:4000/" #(for local testing)
```
Comment (rohanku) [inline client/helper.py:7]: Sous

Comment (oliverye7) [inline client/helper.py]: no :(

File: server/index.js [hunk 2]
```diff
@@ -8,19 +8,19 @@ const app = express();
...
-  }
-  try {
-    await authenticate(req.body.access_token);
-  } catch (error) {
-    res.status(401).end();
-    return;
-  }
-  next();
-});
+// app.use(async (req, res, next) => {
```
Comment (rohanku) [inline server/index.js:11]: Hmmmmm

**Review by oliverye7 — CHANGES_REQUESTED** [review]

**Review by rohanku — CHANGES_REQUESTED** [review]
PR bad

**Review by rohanku — APPROVED** [review]

**Pull request merged**